### ISSUE TITLE
Can't start up a worker in queue mode

### ISSUE THREAD

**Issue by ArkURL** [n8n-io/n8n#8296]
**Describe the bug**
can not start up a new worker when using queue mode.
executed command `./package/cli/bin/n8n worker` but get error message below:
```
$ ./packages/cli/bin/n8n worker
node:internal/modules/cjs/loader:1137
  throw err;
  ^

Error: Cannot find module 'semver'
Require stack:
- D:\dev\cloned\n8n\packages\cli\bin\n8n
    at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
    at Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (D:\dev\cloned\n8n\packages\cli\bin\n8n:22:26)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'D:\\dev\\cloned\\n8n\\packages\\cli\\bin\\n8n' ]
}

Node.js v18.19.0
```

**To Reproduce**
i followed the doc steps: [doc link here](https://docs.n8n.io/hosting/scaling/queue-mode/#set-encryption-key)

1. set encryption key: `export N8N_ENCRYPTION_KEY=<main_instance_encryption_key>`
2. set execution mode: `export EXECUTIONS_MODE=queue`
3. turning on redis service.
4. start a main node in bash window via command `n8n`, successed
5. start a worker node in another bash window via command `./packages/cli/bin/n8n worker`, failed and got error message mentioned above.

**Expected behavior**
Expect the worker node running success and can be schedule by main node.

**Environment (please complete the following information):**

- OS: Window Server 2016
- n8n Version: 1.21.1
- Node.js Version: v18.19.0
- Database system: SQLite 
- Operation mode: queue

**Additional context**
None yet

**Comment by Joffcom** [comment]
Hey @ArkURL,

When running the n8n command for the main instance are you also running from source?

Unless you are making code changes I wouldn't recommend running n8n like this.

It won't solve this problem but don't forget for queue mode you will also need to use Postgres.

**Comment by ArkURL** [comment]
> Hey @ArkURL,
> 
> When running the n8n command for the main instance are you also running from source?
> 
> Unless you are making code changes I wouldn't recommend running n8n like this.
> 
> It won't solve this problem but don't forget for queue mode you will also need to use Postgres.

Thanks for your reply!
I haven't changed the source code since cloning the project to local repository. I think it may relate to `node` or `npm` problem, I'm trying to use pnpm instead to install the packages.

**Comment by ArkURL** [comment]
sadly but use `pnpm` instead of `npm` still not work😥
`pnpm` install the dependencies is working fine and all dependencies were downloaded successfuly. But when I try to run the command below again, a new error occured:
`$ ./packages/cli/bin/n8n work `
error message:
`> Error: command work not found`
And I already set `postgresql` config to make sure `n8n` use `postgresql` as database.

So I tried to use `npm` to run the scripts. But failed again.
My system `node` version is v16.16.0, So I have to use a `node` manager because `n8n` require `node` version is `18` or `20`. But when I used `npm install` to install dependencies, an new error occuied:
```
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "workspace:": workspace:*

npm ERR! A complete log of this run can be found in: D:\cache\npm_cache\_logs\2024-01-12T07_22_45_534Z-debug-0.log
```

Environment:
I use `fnm` as `node` and `npm` manager.
`node` version: Node.js v18.19.0
`npm` version: 10.2.3
